app/services/phone_formatter.rb

```diff
@@ -5,4 +5,11 @@ def self.format(phone, country_code: nil)
     country_code = DEFAULT_COUNTRY if country_code.nil? && !phone&.start_with?('+')
     Phonelib.parse(phone, country_code)&.international
   end
+
+  def self.mask(phone)
+    return '' if phone.blank?
+
+    formatted = Phonelib.parse(phone).national
+    formatted[0..-5].gsub(/\d/, '*') + formatted[-4..-1]
```

Comment:
This could use some fresh eyes -- for now I've just copied the implementation over from PhoneConfiguration. A separate PR/ticket will refresh this

Reply:
it looks good to me. What will the purpose of the refresh be?

Reply:
It has a different interface from the above `format` method -- this one returns an empty string for invalid phone numbers and `format` returns nil. And this one is focused on national numbers whereas `format` has explicit support for international. That's all  I have in mind